pkg/cli/admin/inspect/namespace.go

```diff
@@ -92,3 +101,67 @@ func (o *InspectOptions) gatherNamespaceData(baseDir, namespace string) error {
 	}
 	return nil
 }
+
+// podEligibleForCollection aims to filter out the daemonset pods running on
+// data planes if node count is greater than a threshold,
+// in order to save significant storage size and reduces the collection duration.
+// If these pods need to be collected, recommended path is
+// to use node-selector flag (i.e. --node-selector='node-role.kubernetes.io/worker').
+func (o *InspectOptions) podEligibleForCollection(pod *corev1.Pod) (bool, error) {
+	isDaemonSet := false
+	if len(pod.OwnerReferences) > 0 {
+		for _, OwnerRef := range pod.OwnerReferences {
+			if OwnerRef.Kind == "DaemonSet" {
+				isDaemonSet = true
+				break
+			}
+		}
+	}
+	if !isDaemonSet {
+		return true, nil
+	}
+
+	nodes, err := o.kubeClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
do this once before iterating the pods.  Then use the list of nodes.  If you don't find a node in the list, assume you should collect the pod logs.

Reply:
Updated.